Nodejs/Product/TestAdapterNetStandard/NetCoreTestDiscoverer.cs

```diff
@@ -147,10 +119,76 @@ void ProcessFiles(string fileAbsolutePath)
         }
 
-        private void DiscoverTests(HashSet<string> fileList, TestFramework testFramework, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
+        private void DiscoverTests(Dictionary<string, HashSet<string>> testItems, FrameworkDiscoverer frameworkDiscoverer, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
         {
-            var discoverWorker = new TestDiscovererWorker(projectFullPath, NodejsConstants.ExecutorUri, nodeExePath);
-            discoverWorker.DiscoverTests(fileList, testFramework, logger, discoverySink);
+            foreach (var testFx in testItems.Keys)
+            {
+                var testFramework = frameworkDiscoverer.GetFramework(testFx);
+                if (testFramework == null)
+                {
+                    logger.SendMessage(TestMessageLevel.Warning, $"Ignoring unsupported test framework '{testFx}'.");
+                    continue;
+                }
+
+                var fileList = testItems[testFx];
+
+                var discoverWorker = new TestDiscovererWorker(projectFullPath, NodejsConstants.ExecutorUri, nodeExePath);
+                discoverWorker.DiscoverTests(fileList, testFramework, logger, discoverySink);
+            }
+        }
+
+        private Dictionary<string, HashSet<string>> GetConfigItems(string projectRoot)
+        {
+            var files = Directory.EnumerateFiles(projectRoot, "angular.json", SearchOption.AllDirectories)
+                .Where(x => !x.Contains("\\node_modules\\"));
+            if (files.Any())
+            {
+                return new Dictionary<string, HashSet<string>>()
+                {
+                    {
+                        TestFrameworkDirectories.AngularFrameworkName,
+                        new HashSet<string>(files)
+                    }
+                };
+            }
+
+            return default;
+        }
+
+        private Dictionary<string, HashSet<string>> GetTestItems(string projectRoot, string testFramework, string outDir)
+        {
+            // If we find ts or tsx files, get the JS file and return.
+            var files = Directory.EnumerateFiles(projectRoot, "*.ts?", SearchOption.AllDirectories)
+                .Where(x => !x.Contains("\\node_modules\\"));
+            if (files.Any())
+            {
+                var testFilePaths = files
+                    .Where(x => TypeScriptHelpers.IsTypeScriptFile(x))
+                    .Select(x => TypeScriptHelpers.GetTypeScriptBackedJavaScriptFile(x, projectRoot, outDir));
+
+                return new Dictionary<string, HashSet<string>>()
+                {
+                    {
+                        testFramework,
+                        new HashSet<string>(testFilePaths)
+                    }
+                };
+            }
+
+            files = Directory.EnumerateFiles(projectRoot, "*.js", SearchOption.AllDirectories)
```

Comment:
Not ".jsx"?

Reply:
`.jsx` requires a precompilation process, most likely not understood by this test adapter. I feel like for that scenario we will not be able to assume where the output is going to be.

Let me know if you want to discuss further about this. I could use brainstorm about it as I do believe it might be a valid scenario.

Reply:
I can't say that I've ever used a jsx file myself, but I definitely had the impression that they could be consumed as-is, without pre-processing.  Happy to discuss further.